README.md

```diff
@@ -1 +1,87 @@
-# breakers
\ No newline at end of file
+# Breakers
+
+Breakers is a Ruby gem that implements the circuit breaker pattern for Ruby using a Faraday middleware. It is designed to handle the case
+where your app communicates with one or more backend services over HTTP and those services could possibly go down. Data about the success
+and failure of requests is recorded in Redis, and the gem uses this to determine when an outage occurs. While a service is marked as down,
+requests will continue to flow through every minute to check if it has returned to being alive.
+
+## Installation
+
+Add this line to your application's Gemfile:
+
+```ruby
+gem 'breakers'
+```
+
+And then execute:
+
+    $ bundle
+
+Or install it yourself as:
+
+    $ gem install breakers
+
+## Usage
+
+### Getting Connected
+
+The gem allows you to define your services like this:
+
+```ruby
+service = Breakers::Service.new(name: 'messaging', host: /.*messaging\.va\.gov/, path: /.*/)
+```
+
+The name parameter is used for logging and reporting only. On each request, the host and path will be compared against the requested values
+to see if this service applies.
+
+A Breakers::Client is the basic data structure for accessing the state and creating connections. It requires a redis connection and one or
+more services:
+
+```ruby
+client = Breakers::Client.new(redis_connection: redis, services: [service])
+Breakers.set_client(client)
+```
+
+Now, you can create a new Faraday connection in your code with:
+
+```ruby
+Breakers.new_connection
+```
+
+This method takes optional `url:` and `adapter:` arguments to allow you to customize those values in the connection.
```

Comment:
Can we expose other options, such as the ability to use other middleware and a response parser? This is our current Faraday config for disability claims
```
faraday.options.timeout = DEFAULT_TIMEOUT
        faraday.use      EVSS::ErrorMiddleware
        faraday.use      Faraday::Response::RaiseError
        faraday.response :json, content_type: /\bjson$/
        faraday.adapter  :httpclient
```

Thinking about this a bit more, does it make more sense to have the circuit breaker as a Faraday middleware and not a wrapper?